src/settings/LoanPolicy/components/EditSections/LoansSection/LoansSection.test.js

```diff
@@ -0,0 +1,333 @@
+      container = renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isProfileFixed: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('Should render due date schedule dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        name: 'loansPolicy.fixedDueDateScheduleId',
+        id: 'input_loansPolicy_fixedDueDateSchedule',
+        required: true,
+        children: mockedSchedules,
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(fieldComponentOrderCalls.dueDateSchedule, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.fDDS)).toBeInTheDocument();
+    });
+  });
+
+  describe('Render when "isShortTermLoan" and "isLoanable" return true', () => {
+    let container;
+    const fieldComponentOrderCalls = {
+      closedDueDate: 3,
+    };
+
+    beforeEach(() => {
+      container = renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isShortTermLoan: jest.fn(() => true),
+          isLoanable: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('Should render due date dropdown correctly', () => {
+      const expectedProps = {
+        component: Select,
+        name: 'loansPolicy.closedLibraryDueDateManagementId',
+        children: {
+          options: shortTermLoansOptions,
+          id: messageIds.pleaseSelect,
+        }
+      };
+
+      expect(Field).toHaveBeenNthCalledWith(fieldComponentOrderCalls.closedDueDate, expect.objectContaining(expectedProps), {});
+      expect(container.getByText(messageIds.dueDate)).toBeInTheDocument();
+    });
+  });
+
+  describe('Render when "isOpeningTimeOffsetActive" returns true', () => {
+    beforeEach(() => {
+      renderLoansSection({
+        ...initialProps,
+        policy: {
+          ...mockedPolicy,
+          isOpeningTimeOffsetActive: jest.fn(() => true),
+        }
+      });
+    });
+
+    it('"Period" component should get correct properties for opening time offset field', () => {
+      const expectedProps = {
+        fieldLabel: messageIds.openingTimeOffset,
+        inputValuePath: 'loansPolicy.openingTimeOffset.duration',
+        selectValuePath: 'loansPolicy.openingTimeOffset.intervalId',
+        changeFormValue: initialProps.change,
+        intervalPeriods: {
+          id: messageIds.selectInterval,
+          options: intervalPeriods.slice(0, 2).reverse(),
+        },
+      };
+
+      expect(Period).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+  });
+
+  describe('Updating of component', () => {
+    let container;
+
+    beforeEach(() => {
+      container = renderLoansSection();
+    });
+
+    it('Should not trigger "change" property after component properties updating', () => {
+      container.rerender(<LoansSection {...initialProps} schedules={[]} />);
+      expect(initialProps.change).not.toBeCalled();
```

Comment:
Not sure but it looks like we need to clear this mock `initialProps.change` where we clear `Field` component mock

Reply:
done